prime-router/src/test/kotlin/history/azure/SubmissionFunctionIntegrationTests.kt

```diff
@@ -0,0 +1,541 @@
+package gov.cdc.prime.router.history.azure
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import gov.cdc.prime.router.ActionLog
+import gov.cdc.prime.router.ActionLogLevel
+import gov.cdc.prime.router.FileSettings
+import gov.cdc.prime.router.InvalidParamMessage
+import gov.cdc.prime.router.Receiver
+import gov.cdc.prime.router.Report
+import gov.cdc.prime.router.Sender
+import gov.cdc.prime.router.Topic
+import gov.cdc.prime.router.azure.DataAccessTransaction
+import gov.cdc.prime.router.azure.DatabaseAccess
+import gov.cdc.prime.router.azure.MockHttpRequestMessage
+import gov.cdc.prime.router.azure.WorkflowEngine
+import gov.cdc.prime.router.azure.db.enums.TaskAction
+import gov.cdc.prime.router.azure.db.tables.pojos.Action
+import gov.cdc.prime.router.azure.db.tables.pojos.ItemLineage
+import gov.cdc.prime.router.azure.db.tables.pojos.ReportFile
+import gov.cdc.prime.router.azure.db.tables.pojos.ReportLineage
+import gov.cdc.prime.router.common.JacksonMapperUtilities
+import gov.cdc.prime.router.common.UniversalPipelineTestUtils
+import gov.cdc.prime.router.db.ReportStreamTestDatabaseContainer
+import gov.cdc.prime.router.db.ReportStreamTestDatabaseSetupExtension
+import gov.cdc.prime.router.history.DetailedSubmissionHistory
+import gov.cdc.prime.router.tokens.AuthenticatedClaims
+import gov.cdc.prime.router.tokens.AuthenticationType
+import gov.cdc.prime.router.tokens.oktaSystemAdminGroup
+import gov.cdc.prime.router.unittest.UnitTestUtils
+import io.mockk.every
+import io.mockk.mockkObject
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.testcontainers.junit.jupiter.Testcontainers
+import java.time.OffsetDateTime
+import java.util.UUID
+
+@Testcontainers
+@ExtendWith(ReportStreamTestDatabaseSetupExtension::class)
+class SubmissionFunctionIntegrationTests {
+
+    class ReportNodeBuilder {
```

Comment:
Leaving this here for now and it can get moved once another test wants similar setup